.github/workflows/branch--lint-unit-and-smoke-test.yml

```diff
@@ -1,16 +1,68 @@
-name: Smoke Tests
+name: Branch QA
 on:
   pull_request:
     branches:
       - master
   workflow_dispatch:
 jobs:
-  test:
+  lint-and-test:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        app: [designer, runner, model]
+    name: lint-and-test-${{matrix.app}}
     steps:
       - uses: actions/checkout@v2
+      - name: Use Node.js
+        uses: actions/setup-node@v1
         with:
-          fetch-depth: 0
+          node-version: "12.x"
+
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn config get cacheFolder)"
+
+      - name: Configure yarn caching
+        uses: actions/cache@v2
+        id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      - name: Install dependencies
+        run: yarn install
+
+      - name: Build dependencies
+        run: yarn build:dependencies
```

Comment:
minor, model will build itself here - it might add a few s to the model pipeline but no big deal

Reply:
Without it `test-cov` fails

```
Error requiring file: .../digital-form-builder/designer/client/components/ComponentCreate/__tests__/ComponentCreateList.test.tsx
Cannot find module '.../digital-form-builder/node_modules/@xgovformbuilder/model/dist/module/index.js'. 
```

Reply:
I mean, the model doesn't need to build itself for tests to pass since we use babel runtime (iirc?) - but designer/runner will need it. I'll resolve it's nbd